### PR TITLE
Build: Introduce shellcheck

### LINKED ISSUE #16198: Consider to use shellcheck for bash lint
**What would you like to be added**:
We have a good number of shell scripts used for various tasks. According to github stats it's 1.1% (currently) of our source. 

It seems it might be a good idea to lint that logic?

**Why is this needed**:
Implement [shellcheck](https://www.shellcheck.net/) as a part of your CI run (and it would be cool if you could run it locally of course)

### PR DESCRIPTION
**Pull request by markelog**
Fixes #16198

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -127,6 +127,15 @@ jobs:
           name: Lint Go
           command: 'make lint-go'
 
+  shellcheck:
+    machine: true
```
Comment (markelog) [inline .circleci/config.yml]: Here, I wanted to use the same command make command as it intendant to be used locally

File: scripts/circle-test-backend.sh [hunk 2]
```diff
@@ -1,20 +1,12 @@
...
-    echo "Executing '$command'"
-    eval $command
-    rc=$?
-    if [ $rc -ne 0 ]; then
-        echo "'$command' returned $rc."
-        exit $rc
-    fi
-}
+
+source "$(dirname "$0")/helpers/exit-if-fail.sh"
```
Comment (xlson) [inline scripts/circle-test-backend.sh]: Been meaning to get this fixed for ages. Great to finally get this one taken care of!

**Review by xlson — APPROVED** [review]
Great work Oleg!

File: scripts/build/prepare-enterprise.sh [hunk 3]
```diff
@@ -11,5 +10,5 @@ fi
 
 git clone -b "$_target" --single-branch git@github.com:grafana/grafana-enterprise.git --depth 1
 
-cd grafana-enterprise
+cd grafana-enterprise || exit
```
[the remark this answers is not in the document]
Comment (xlson) [reply scripts/build/prepare-enterprise.sh]: Surprised there weren't more cases found of not failing at the right time, but perhaps we need into the scripts manually to make sure all have been found.

File: Makefile [hunk 4]
```diff
@@ -111,6 +112,9 @@ go-vet:
 
 lint-go: go-vet golangci-lint revive revive-alerting gosec
 
+shellcheck:
+	docker run --rm -v "$$PWD:/mnt" koalaman/shellcheck:stable $(SH_FILES)
```
[the remark this answers is not in the document]
Comment (markelog) [reply Makefile]: In makefile, if you want to escape the dollar sign you have to add another dollar sign :), which is what we want here

File: scripts/build/prepare-enterprise.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (markelog) [reply scripts/build/prepare-enterprise.sh]: Good idea! I will check it out and will keep an eye on it

File: scripts/circle-test-backend.sh (hunk 2 above)
Comment (markelog) [reply scripts/circle-test-backend.sh]: :))

File: Makefile (hunk 4 above)
[the remark this answers is not in the document]
Comment (xlson) [reply Makefile]: That's good to know :)

File: scripts/build/prepare-enterprise.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (markelog) [reply scripts/build/prepare-enterprise.sh]: Updated!

File: Makefile [hunk 5]
```diff
@@ -4,6 +4,7 @@
 
 GO = GO111MODULE=on go
 GO_FILES ?= ./pkg/...
+SH_FILES ?= $(shell find ./scripts -name *.sh)
```
Comment (markelog) [inline Makefile]: @aocenas you might be interested in this solution

Comment (aocenas) [reply Makefile]: In the end I went with this:
```
# recursive wildcard https://stackoverflow.com/a/12959764/1835885
rwildcard=$(wildcard $1$2) $(foreach d,$(wildcard $1*),$(call rwildcard,$d/,$2))
```

Not sure if important (I think that ship already sailed) but it should also work on Win as it does not use shell.

**Comment by xlson** [comment]
@markelog can we merge this?

**Comment by markelog** [comment]
Just to note, I think we have definite confidence that those scripts work only after we run them, so if there is any issues with grafana release this pull could be a culprit.

That being said, I think we can land this!

**Pull request merged**